module/VuFind/src/VuFind/View/Helper/Root/HelpText.php

```diff
@@ -100,39 +82,29 @@ public function getWarnings()
      *
      * @return string|bool
      */
-    public function render($name, $context = null)
+    public function render($name, $context=[])
     {
-        // Set up the needed context in the view:
-        $this->contextHelper->__invoke($this->getView());
-        $oldContext = $this->contextHelper
-            ->apply(null === $context ? [] : $context);
-
         // Sanitize the template name to include only alphanumeric characters
         // or underscores.
         $safe_topic = preg_replace('/[^\w]/', '', $name);
 
-        // Clear warnings
         $this->warnings = [];
+        $html = $this->contentHelper->renderTranslated(
+            $safe_topic,
+            'HelpTranslations',
+            $context,
+            $pageDetails,
+            '%pathPrefix%/%language%/%pageName%',
+        );
 
-        $resolver = $this->getView()->resolver();
-        $tpl = "HelpTranslations/{$this->language}/{$safe_topic}.phtml";
-        if ($resolver->resolve($tpl)) {
-            $html = $this->getView()->render($tpl);
-        } else {
-            // language missing -- try default language
-            $tplFallback = 'HelpTranslations/' . $this->defaultLanguage . '/'
-                . $safe_topic . '.phtml';
-            if ($resolver->resolve($tplFallback)) {
-                $html = $this->getView()->render($tplFallback);
-                $this->warnings[] = 'Sorry, but the help you requested is '
-                    . 'unavailable in your language.';
-            } else {
-                // no translation available at all!
-                $html = false;
-            }
+        if (!$html) {
+            $this->warnings[] = 'Sorry, but the help you requested is '
```

Comment:
When testing, I never saw this message. In [the template](https://github.com/vufind-org/vufind/blob/dev/themes/root/templates/help/home.phtml), when $html is false, the warnings are not displayed at all, and the `help_page_missing` string is displayed instead. Perhaps it is worth changing the message here to `help_page_missing` (so we don't need to translate a new message for i18n), and then simplifying the template to take advantage of the logic here.

Reply:
The message has been changed in the meantime, but i'm not sure how exactly you think the template could be simplified. Could you please explain?

Reply:
I thought you might be able to simplify to:

<pre>
<?php $this->headTitle($this->translate('Help')); ?>
<?php foreach ($this->helpText()->getWarnings() as $warning): ?>
  <p class="warning"><?=$this->transEsc($warning)?></p>
<?php endforeach; ?>
<?=$this->helpText()->render($topic)) ?>
</pre>

This way, you leverage the existing warnings mechanism for the "no help" message instead of a special case, and you don't bother checking the return value of render() since it will either return something or nothing, and echoing either will be okay as long as the warning is displayed correctly.

Reply:
The thing is => the way the HelpText view helper is implemented right now, getWarnings just outputs the messages found in render. So this would imply that you pass the $topic also to getWarnings() and either 2 lookups being made, or the lookup results would need to be cached by $topic inside the view helper (or even deeper). Is this what you mean?

Reply:
@mtrojan-ub, you're right, that won't work... but it could be adjusted to:

```
<?php $this->headTitle($this->translate('Help')); ?>
<?php $helpText = $this->helpText()->render($topic)); ?>
<?php foreach ($this->helpText()->getWarnings() as $warning): ?>
  <p class="warning"><?=$this->transEsc($warning)?></p>
<?php endforeach; ?>
<?=$helpText?>
```

That's still a bit simpler than the existing template.